tests/commands_test.py

```diff
@@ -370,44 +370,83 @@ def test_touch_cmd(
     memcache_socket.get_response.reset_mock()
 
 
+def assert_called_once_with_command(mock, expected_cmd, **expected_kwargs):
+    assert_called_with_commands(mock, [expected_cmd], **expected_kwargs)
+
+
+def assert_called_with_commands(mock, expected_cmds, **expected_kwargs):
+    def split_cmd(cmd: bytes):
+        assert cmd.endswith(b"\r\n"), f"Unexpected cmd format: {cmd}"
+        pieces = cmd[:-2].split(b" ")
+        command = pieces.pop(0)
+        key = pieces.pop(0)
+        return command, key, set(pieces)
+
+    calls = mock.call_args_list
+    assert len(calls) == len(
+        expected_cmds
+    ), f"Expected exactly {len(expected_cmds)} calls to {mock}"
+    for i in range(len(calls)):
+        args, kwargs = calls[i]
+        expected_cmd = expected_cmds[i]
+        assert len(args) == 1, f"Unexpected num of args to {mock}"
+        assert (
+            kwargs == expected_kwargs
+        ), f"Unexpected kwargs to {mock}: {kwargs} expected {expected_kwargs}"
+        cmd = args[0]
+        actual_cmd, actual_key, actual_flags = split_cmd(cmd)
+        expected_cmd, expected_key, expected_flags = split_cmd(expected_cmd)
+        assert (
+            actual_cmd == expected_cmd
+        ), f"Unexpected cmd to {mock}: {actual_cmd} expected {expected_cmd}"
+        assert (
+            actual_key == expected_key
+        ), f"Unexpected key to {mock}: {actual_key} expected {expected_key}"
+        assert (
+            actual_flags == expected_flags
+        ), f"Unexpected flags to {mock}: {actual_flags} expected {expected_flags}"
+
+
 def test_get_cmd(memcache_socket: MemcacheSocket, cache_client: CacheClient) -> None:
     memcache_socket.get_response.return_value = Miss()
 
     cache_client.get(key="foo")
-    memcache_socket.sendall.assert_called_once_with(
-        b"mg foo t l v h f\r\n", with_noop=False
+    assert_called_once_with_command(
+        memcache_socket.sendall, b"mg foo t l v h f\r\n", with_noop=False
```

Comment:
This change was done because python 3.11 changes order of sets, so the command generated was not deterministic. I considered sorting on the client, to make it deterministic, but that would add some perf hit to production for the sake of deterministic behavior and tests. We now split the command and compare sets of the flags during tests, so order is not important.